### PR TITLE
V1

### PR DESCRIPTION
**Pull request by void-star-star**
First version of a python based Web crawler that finds email addresses.

### CODE REVIEW

File: find_email_address.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    driver = webdriver.Chrome()
+else:
+    driver = webdriver.Remote(desired_capabilities=webdriver.DesiredCapabilities.HTMLUNIT)
+
+def log(msg):
+    if args.verbose:
+        print(msg)
+
+def find_links(driver, domain):
+    retval = {'emails': set(), 'links': set()} # emails for mailto: links, links for everything else
```
Comment (clee-jana) [inline find_email_address.py:37]: any particular reason you chose a function with this return type over some object that had the emails and links as a property on that object?

File: find_email_address.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            continue
+        url = urlparse(link)
+
+        if url.scheme.startswith('mailto'):
+            log('Email found: ' + url.path)
+            retval['emails'].add(url.path)
+        else:
+            #pprint(url)
+
+            if url.netloc.startswith(domain):
```
Comment (clee-jana) [inline find_email_address.py:53]: why `startswith` instead of `==`?

File: find_email_address.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+parser.add_argument('hostname', metavar='domain', nargs=1,
+                    help='Host to scrape')
+parser.add_argument('--port',
+                    help='Port number, default: 80')
+parser.add_argument('--chrome',
+                    help='Use Chrome instead of HTMLUNIT',  action="store_true")
+parser.add_argument('--verbose',
+                    help='Output additional logging',  action="store_true")
+#parser.add_argument('--depth',
+#                    help='Unimplemented: Max number of pages to fetch, default: 100')
```
Comment (clee-jana) [inline find_email_address.py:16]: did you mean this to be depth or just a max number of pages?

File: find_email_address.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+if args.chrome:
+    driver = webdriver.Chrome()
+else:
+    driver = webdriver.Remote(desired_capabilities=webdriver.DesiredCapabilities.HTMLUNIT)
+
+def log(msg):
+    if args.verbose:
+        print(msg)
+
+def find_links(driver, domain):
```
Comment (clee-jana) [inline find_email_address.py:36]: in my opinion this is a bit of a misnomer because it's finding both links and emails.

**Review by clee-jana — COMMENTED** [review]
responding over email

File: find_email_address.py (hunk 3 above)
Comment (void-star-star) [reply find_email_address.py:16]: I meant it to be max pages, renamed in the new PR.

File: find_email_address.py (hunk 4 above)
Comment (void-star-star) [reply find_email_address.py:36]: Agreed, and re-arranged in next PR.

File: find_email_address.py (hunk 2 above)
Comment (void-star-star) [reply find_email_address.py:53]: Whoops, I hadn't read far enough in the documentation for urlparser actually.  New PR uses url.hostname.  netloc included port number so used startwith.

**Comment by void-star-star** [comment]
Going to close and open v2 after a half hour of frantic refactoring.

**Pull request closed without merging**